### PR TITLE
fix: fixed an issue that prevented clearing tags for releases

### PR DESCRIPTION
**Pull request by mzedel**
Ticket: None
Changelog: None

### CODE REVIEW

**Comment by mzedel** [comment]
@mender-test-bot start pipeline --fast

File: model/release.go [hunk 1]
```diff
@@ -65,19 +64,15 @@ func (tags Tags) MarshalJSON() ([]byte, error) {
...
-	s := *tags
-	sort.Slice(s, func(i, j int) bool {
-		return s[i] < s[j]
-	})
-	i, j := 0, 0
-	for j < len(s) {
-		if s[i] != s[j] {
-			i++
-			s[i] = s[j]
+	set := make(map[Tag]bool)
```
Comment (alfrunes) [inline model/release.go:67]:
```suggestion
	set := make(map[Tag]struct{})
```

File: model/release.go [hunk 2]
```diff
@@ -65,19 +64,15 @@ func (tags Tags) MarshalJSON() ([]byte, error) {
...
+	result := []Tag{}
+	for _, item := range *tags {
+		if _, exists := set[item]; !exists {
+			set[item] = true
```
Comment (alfrunes) [inline model/release.go:71]:
```suggestion
			set[item] = struct{}{}
```

File: model/release.go [hunk 3]
```diff
@@ -65,19 +64,15 @@ func (tags Tags) MarshalJSON() ([]byte, error) {
...
-	sort.Slice(s, func(i, j int) bool {
-		return s[i] < s[j]
-	})
-	i, j := 0, 0
-	for j < len(s) {
-		if s[i] != s[j] {
-			i++
-			s[i] = s[j]
+	set := make(map[Tag]bool)
+	result := []Tag{}
```
Comment (alfrunes) [inline model/release.go:68]:
```suggestion
	result := Make(Tags, 0, len(*tags))
```

**Review by alfrunes — COMMENTED** [review]
Couple of suggestions for pruning additional allocations.
I also fixed the previous implementation here https://github.com/mzedel/deployments/pull/7. But I'm not sure if readability would be an issue...

**Comment by mzedel** [comment]
@mender-test-bot start pipeline --fast

**Review by alfrunes — APPROVED** [review]

**Pull request merged**